client/src/Components/Spectrogram/AnnotationViewer.jsx

```diff
@@ -25,17 +29,18 @@ const AnnotationViewer = (props) => {
 
     // Now update the actual meta.annotations
     const f = annotations[annot_indx]['index']; // remember there are 2 different indexes- the ones on the screen and the meta.annotations
-    let updatedAnnotations = [...props.meta.annotations];
+    let updatedAnnotations = [...meta.annotations];
+    const updatedAnnotation = { ...updatedAnnotations[f] };
```

Comment:
Should this be a constant when you are making changes to it further diwn?

Reply:
Good catch will fix.